build.gradle

```diff
@@ -582,6 +583,48 @@ configure(updateBuildScript) {
     description = 'Updates the build script to the latest version'
 }
 
+// Deobfuscation
+
+def deobf(String sourceURL, String fileName) {
+    return deobf(sourceURL, fileName, false)
+}
+
+def deobf(String sourceURL, String fileName, boolean notch) {
+    def bon2File  = "$projectDir/data/BON2-2.5.0.jar"
+    def cacheFile = "$projectDir/deobfDependencies/" + fileName + ".jar"
+    def deobfFile = "$projectDir/deobfDependencies/" + fileName + "-deobf.jar"
+
+    if(file(deobfFile).exists()) {
+        return files(deobfFile)
+    }
+
+    download {
+        src 'https://github.com/glowredman/BON2/releases/download/2.5.0/BON2-2.5.0.CUSTOM-all.jar'
+        dest bon2File
+        quiet true
+        overwrite false
+    }
+
+    download {
+        src sourceURL
+        dest cacheFile
+        quiet true
+        overwrite false
+    }
+
+    if(notch) {
+        exec {
+            commandLine 'java', '-jar', bon2File, '--inputJar', cacheFile, '--outputJar', deobfFile, '--mcVer', '1.7.10', '--mappingsVer', 'stable_12'
+        }
+    } else {
+        exec {
+            commandLine 'java', '-jar', bon2File, '--inputJar', cacheFile, '--outputJar', deobfFile, '--mcVer', '1.7.10', '--mappingsVer', 'stable_12', '--notch'
```

Comment:
Is this —notch if … false? Should it be inverted?

Reply:
True... Why did my testing still work? If it works with --notch no matter how the original jar is compiled, I can probably simplfy this a bit

Reply:
Nice. Kick it up a notch. (Maybe the deobf was already there?)

Reply:
Nah, if I comment the whole deobf part out, the build fails with `Could not find method deobf() for arguments [https://www.immibis.com/mcmoddl/files/redlogic-59.1.13.jar, redlogic-59.1.13] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.`